src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -69,18 +69,49 @@ on_exit() {
     set -xe
   fi
 
+  # update project nextflow history
+  update_project_history
+
+  # backup cache
+  if [[ $no_future_resume == false ]]; then
+    echo "=== Execution complete — uploading Nextflow cache and history file"
+    # only upload cache.tar(cache and history)
+    # files in workdir $DX_PROJECT_CONTEXT_ID:/nextflow_cache_db/$NXF_UUID/work/ are uploaded by the plugin after each subjob
+
+    # wrap cache folder and history and upload cache.tar
+    if [[ -n "$(ls -A .nextflow)" ]]; then
+      tar -cf cache.tar .nextflow
+      # remove any existing cache.tar with the same session id
```

Comment:
Is it possible that two jobs have the same session ID? The resumed job has the same session ID, correct? If we run two jobs with --resume <same_session_id> then we might run into a conflict, where for example job A uploads cache.tar, the other job B deletes it and uploads its own cache.tar and it will incorrectly seem this tar is from this execution (job A). Not 100% sure such a scenario is possible.

Reply:
Yes all resumed jobs will have the same session id.

Yes if we run job A and B like you said, B will indeed delete cache.tar uploaded by job A (if it ends after job A) and upload its own one. But since cache stores the info to resume previous work files, 
- if B proceeds further than A, we should update the cache, which will have more info
- if B ends at the same stage as A or even ends earlier, what in A's cache.tar (which is downloaded to B's container) will still be included in B's cache.tar.

And since the cache.tar is always found using its path `project-id:nextflow_cache_db/<session id>/cache.tar` but not any job id, if the session is resumed again by job C, it would find this cache, and it should not matter if this tar is from A or B, because it always shall contains the most info of this session.